### PR TITLE
Documentation for OffsetTime and OffsetDate

### LINKED ISSUE #864: Consider OffsetTime and OffsetDate types
Not saying per se that I want this, but there may be a need for an `OffsetTime` type - that is, a type that is `LocalTime` + `Offset`.

https://stackoverflow.com/questions/44514504/offsettime-in-nodatime

ISO8601 has this in section *4.2.5.2 Local time and the difference from UTC*, supporting the same basic/extended formats that we are used to with `OffsetDateTime`, sans the date portion.

Java 8 has this with [`java.time.OffsetTime`](https://docs.oracle.com/javase/8/docs/api/java/time/OffsetTime.html).

PostgreSQL has this with [`time with time zone`](https://www.postgresql.org/docs/current/static/datatype-datetime.html), and mentions that it's in the SQL standard (ISO 9075), but discourages its use.  From their docs:

> The type `time with time zone` is defined by the SQL standard, but the definition exhibits properties which lead to questionable usefulness. In most cases, a combination of date, time, timestamp without time zone, and timestamp with time zone should provide a complete range of date/time functionality required by any application.

Do we want to add this to Noda Time?

### PR DESCRIPTION
**Pull request by jskeet**
Minimal, but enough to specify text formats etc.

Fixes #864 - and unblocks a 2.3.0 release.

### CODE REVIEW

File: src/NodaTime.Web/Markdown/unstable/core-types.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
+@Title="Core types quick reference"
```
Comment (malcolmr) [inline src/NodaTime.Web/Markdown/unstable/core-types.md]: Sigh, git doesn't record where copies come from, so I had to clone the patch locally and run diff with `--find-copies-harder` to get a good diff.

File: src/NodaTime.Web.Test/NodaTime.Web.Test.csproj [hunk 2]
```diff
@@ -14,5 +14,9 @@
     <PackageReference Include="NUnit" Version="3.6.1" />
     <PackageReference Include="NUnitLite" Version="3.6.1" />
   </ItemGroup>
+
+  <ItemGroup>
+    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
```
Comment (malcolmr) [inline src/NodaTime.Web.Test/NodaTime.Web.Test.csproj]: (This is https://github.com/Microsoft/vstest/issues/472, and while there's a lot of discussion there about whether this makes sense, it's probably fine to leave it as-is.)

File: src/NodaTime.Web/NodaTime.Web.csproj [hunk 3]
```diff
@@ -24,7 +24,6 @@
     <PackageReference Include="morelinq" Version="2.10.0" />
   
     <!-- Handling of non-code folders -->
-    <Folder Include="Markdown\" />
```
Comment (malcolmr) [inline src/NodaTime.Web/NodaTime.Web.csproj]: I assume this was redundant with the below?

File: src/NodaTime.Web.Test/NodaTime.Web.Test.csproj (hunk 2 above)
Comment (jskeet) [reply src/NodaTime.Web.Test/NodaTime.Web.Test.csproj]: Ah, missed this one. I removed it again from all the rest of the projects before merging. Think I'll do so again here.

File: src/NodaTime.Web/NodaTime.Web.csproj (hunk 3 above)
Comment (jskeet) [reply src/NodaTime.Web/NodaTime.Web.csproj]: Apparently so. I didn't do that myself, so presumably VS was just being clever...

**Pull request merged**